### PR TITLE
Ensure both property and value exist for Blob's HTTP proxy

### LINKED ISSUE #12187: Apollo client not working due to "Invalid response for blob"
### Problem Description

There seems to be an issue using the `@apollo/client` library. When that library is added, running the app without the web debugger results in errors whenever `fetch()` is called or something from the `@apollo/client` is imported. Everything works fine via the web debugger and also after removing the apollo client library.

The issue might be related to https://github.com/microsoft/react-native-windows/issues/11439 but should be resolved in the latest RNW versions. I've tested this with both RNW 0.71.38 & 0.72.9 and got the same errors. We've upgraded from version 0.69.6, where everything worked fine.

### Steps To Reproduce

1. set up a new app following the RNW "Get Started" guide
2. add `@apollo/client` and `graphql` dependencies following their guides
3. create a simple apollo wrapper following their guides
4. open the app and disable the web debugger
5. the app shows a blank page and there are multiple errors in the console:
`Error: Invalid response for blob` and `Error: ENOENT: no such file or directory, open '\\?\E:\git\rnw-apollo\code'`

### Expected Results

The app loads normally, renders the UI, and can fetch data

### CLI version

11.3.7

### Environment

```markdown
System:
  OS: Windows 10 10.0.19045
  CPU: "(12) x64 AMD Ryzen 5 3600 6-Core Processor              "
  Memory: 14.48 GB / 31.95 GB
Binaries:
  Node:
    version: 20.7.0 (-> also tested with v18 & v16)
    path: C:\Program Files\nodejs\node.EXE
  Yarn:
    version: 1.22.5
    path: C:\Program Files (x86)\Yarn\bin\yarn.CMD
  npm:
    version: 10.1.0
    path: C:\Program Files\nodejs\npm.CMD
  Watchman:
    version: 20200816.222424.0
    path: C:\Users\XXX\AppData\Local\watchman\watchman.EXE
SDKs:
  Android SDK: Not Found
  Windows SDK:
    AllowDevelopmentWithoutDevLicense: Enabled
    AllowAllTrustedApps: Enabled
    Versions:
      - 10.0.14393.0
      - 10.0.17134.0
      - 10.0.17763.0
      - 10.0.18362.0
      - 10.0.19041.0
      - 10.0.22621.0
IDEs:
  Android Studio: Not Found
  Visual Studio:
    - 17.7.34031.279 (Visual Studio Community 2022)
    - 16.11.32126.315 (Visual Studio Community 2019)
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-windows:
    installed: 0.72.9
    wanted: 0.72.9
npmGlobalPackages:
  "*react-native*": Not Found
```

### Target Platform Version

None

### Target Device(s)

Desktop

### Visual Studio Version

None

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

Here is a basic repository to reproduce the error: https://github.com/gtRfnkN/rnw-apollo

Start the app via `yarn windows` . The app should start normally and show 2 buttons: one to load an apollo component and another to trigger a `fetch()`. Clicking either button throws errors.

### PR DESCRIPTION
**Pull request by JunielKatarn**
- Check HTTP handler property and value exist
- Change files

## Description
Ensure HTTP proxy (handler) is correctly registered by Blob resource after instance reload.

### Type of Change
- Bug fix (non-breaking change which fixes an issue)

### Why

When direct debugging is used instead of web debugging, module initialization happens out of order and HTTP blob responses get mishandled due to the lack of Blob module (and associated resources) setup.

This was partially addressed by #11439.

On React Instance reload, the property bag entries `Http.Proxy` and `Blob.Resource` still exist but their values are no longer valid (they were discarded next to the previous instance's module hierarchy).

Therefore, there is no HTTP proxy that can register the Blob resource nor a Blob resource that can register the HTTP proxy inside their corresponding modules, leading to the `Invalid response for blob` exception.

Resolves #12187

### What

This change ensures both the entry **and** value for `Http.Proxy` exist in the React Instance's property bag, else it tries the best-effort approach used for #11439.

## Testing
See sample repository `jurocha-ms/Repro` under path `ReactNative/blobInvReload` and follow on-screen instructions.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12417)

### CODE REVIEW

**Comment by acoates-ms** [comment]
Shouldn't we be clearing the value of these properties specifically during instance shutdown.  -- That way we can be sure that none of the values are being polluted with previous values between the instance reloads.  We should always get an instance destroy event before the new instance is started.

**Comment by JunielKatarn** [comment]
> Shouldn't we be clearing the value of these properties specifically during instance shutdown. -- That way we can be sure that none of the values are being polluted with previous values between the instance reloads. We should always get an instance destroy event before the new instance is started.

As discussed offline, this enhancement will applied in a follow-up PR.

**Review by acoates-ms — APPROVED** [review]

File: vnext/Shared/Networking/DefaultBlobResource.cpp [hunk 1]
```diff
@@ -140,15 +140,19 @@ void DefaultBlobResource::Release(string &&blobId) noexcept /*override*/ {
...
-    // #11439 - The absence of HttpModule.Proxy may be caused by a module initialization race condition.
-    // Best-effort approach to set up the request/response handlers by exposing this interface to dependents
-    // (i.e. IHttpResource).
+  }
+
+  // #11439 - The absence of HttpModule.Proxy may be caused by a module initialization race condition.
+  // Best-effort approach to set up the request/response handlers by exposing this interface to dependents
+  // (i.e. IHttpResource).
+  if (!handlerAdded) {
     auto propId = msrn::ReactPropertyId<msrn::ReactNonAbiValue<weak_ptr<IBlobResource>>>{L"Blob.Resource"};
```
Comment (vmoroz) [inline vnext/Shared/Networking/DefaultBlobResource.cpp:156]: It is better to keep all property ids as static variables. This way we do not have to re-create them for each call.

**Review by vmoroz — APPROVED** [review]

File: vnext/Shared/Networking/DefaultBlobResource.cpp (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Shared/Networking/DefaultBlobResource.cpp:156]: @acoates-ms has mentioned this before.

I will add  follow-up issue to apply this throughout all the networking-related classes.

**Pull request merged**